apps/site/src/pages/account/api-keys-panel.tsx

```diff
@@ -1,142 +1,116 @@
+import { PanelSection } from "../../components/pages/account/panel-section";
 import { apiClient } from "../../lib/api-client";
+import { ApiKeysContext } from "./api-keys-panel/api-keys-context";
+import { ApiKeysEmptyState } from "./api-keys-panel/api-keys-empty-state";
+import { ApiKeysList } from "./api-keys-panel/api-keys-list";
+import {
+  ApiKeyProps,
+  ApiKeysContextValue,
+  KeyActionStatus,
+} from "./api-keys-panel/types";
 
 export const ApiKeysPanel: FunctionComponent = () => {
-  const [activeApiKeys, setActiveApiKeys] = useState<
-    UserFacingApiKeyProperties[]
-  >([]);
+  const [apiKeys, setApiKeys] = useState<ApiKeyProps[]>([]);
+  const [keyActionStatus, setKeyActionStatus] = useState<KeyActionStatus>();
+  const [isCreatingNewKey, setIsCreatingNewKey] = useState(false);
+  const [newlyCreatedKeyIds, setNewlyCreatedKeyIds] = useState<string[]>([]);
 
-  const [generateKeyStatus, setGenerateStatus] = useState<{
-    showModal: boolean;
-    keyToRegenerate?: UserFacingApiKeyProperties | undefined;
-  }>({
-    showModal: false,
-  });
+  const fetchAndSetApiKeys = useCallback(async () => {
+    const { data } = await apiClient.getUserApiKeys();
 
-  const fetchAndSetApiKeys = () =>
-    apiClient
-      .getUserApiKeys()
-      .then(({ data }) =>
-        data
-          ? setActiveApiKeys(
-              data.apiKeysMetadata.filter((key) => !key.revokedAt),
-            )
-          : null,
-      );
+    if (data) {
+      setApiKeys(data.apiKeysMetadata.filter((key) => !key.revokedAt));
+    }
+  }, []);
 
   useEffect(() => {
-    /** @todo handle errors and show the user a msg */
     void fetchAndSetApiKeys();
+    /** this useEffect meant to be only run once */
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
-  const tableRows: TableRows = useMemo(
-    () =>
-      activeApiKeys.map((key) => [
-        key.displayName,
-        key.publicId,
-        key.lastUsedAt ? (
-          <DateTimeCell key="lastUsed" timestamp={key.lastUsedAt} />
-        ) : (
-          "Never"
-        ),
-        <DateTimeCell key="createdAt" timestamp={key.createdAt} />,
-      ]),
-
-    [activeApiKeys],
+  const apiKeysContextValue: ApiKeysContextValue = useMemo(
+    () => ({
+      apiKeys,
+      setApiKeys,
+      isCreatingNewKey,
+      setIsCreatingNewKey,
+      newlyCreatedKeyIds,
+      setNewlyCreatedKeyIds,
+      keyActionStatus,
+      setKeyActionStatus,
+      fetchAndSetApiKeys,
+    }),
+    [
+      apiKeys,
+      setApiKeys,
+      isCreatingNewKey,
+      setIsCreatingNewKey,
+      newlyCreatedKeyIds,
+      setNewlyCreatedKeyIds,
+      keyActionStatus,
+      setKeyActionStatus,
+      fetchAndSetApiKeys,
+    ],
   );
-  const activeKey = activeApiKeys[0];
 
-  const closeGenerateModal = () => {
-    setGenerateStatus({
-      showModal: false,
-      keyToRegenerate: undefined,
-    });
-  };
+  const shouldRenderEmptyState = !apiKeys.length && !isCreatingNewKey;
```

Comment:
This means it renders the empty state while the api keys are loading, which is maybe not ideal – probably want a skeleton or some other loader